spec/models/task_spec.rb

```diff
@@ -44,12 +44,26 @@
       expect(task.reload).to be_completed
     end
 
-    it 'records the user who submitted it' do
+    it 'records the user who created the submission' do
       task.file_no_business!(user)
       submission = task.latest_submission
 
       expect(submission.created_by).to eq(user)
     end
+
+    it 'records the user who completed the submission' do
+      task.file_no_business!(user)
+      submission = task.latest_submission
+
+      expect(submission.submitted_by).to eq(user)
+    end
+
+    it 'records the submission time' do
+      task.file_no_business!(user)
+      submission = task.latest_submission
+
+      expect(submission.submitted_at).to be_present
```

Comment:
Can we make this spec better by verifying the actual value and not just presence? We can use a `be_within` matcher to approximate-match the timestamp, or use the ActiveSupport testing time helpers to freeze time and match against that.

Reply:
Ooh, things have really moved on since Timecop was the only time traveller in town!